### ISSUE TITLE
Compatibility problems with upcoming Eclipse 2021-03

### ISSUE THREAD

**Issue by jonahgraham** [Genuitec/CodeTogether#271]
Try installing CodeTogether in Eclipse JEE 2021-03 M3. More details are in Eclipse bugtracker https://bugs.eclipse.org/bugs/show_bug.cgi?id=571323. While there are problems on the Eclipse side, you can mitigate those problems by using Gson in a way that is less likely to interfere with others.

In particular, see Comment 14. It looks to me like the use of x-friends on the exports is not doing what you are intending it to do. It looks like the intention is for the code together core bundle to only expose com.google.gson to other code together bundles. But how it is written now, com.google.gson can be resolved for all bundles.

Good news is 2021-03 will include a newer version of com.google.gson, so hopefully you can simply stop having a copy in your code.

Additionally, this same thing probably applies to the other packages exported in your bundle that aren't in your namespace.

**Comment by wgalanciak** [comment]
Jonah, thanks for reporting and detailed analysis. We are going to work on it as soon as possible.

**Comment by TimWebb** [comment]
Jonah, thanks for raising this.  We ended up with this solution as part of finding a balance for working with older versions of Eclipse.  We support all the way down to Neon and quite a large group of users running on earlier versions.  Short of running on a different package namespace (rebuilding the jars), this becomes a bit complicated.  We found shipping a newer version of the plugin broke a number of things in earlier versions of Eclipse that depended on Gson by require-bundle.  As Wojtek mentioned, we will see what we can do here...
